src/sources/internal_logs.rs

```diff
@@ -43,15 +49,56 @@ pub struct InternalLogsConfig {
 
 impl_generate_config_from_default!(InternalLogsConfig);
 
+impl InternalLogsConfig {
+    /// Generates the `schema::Definition` for this component.
+    fn schema_definition(&self, log_namespace: LogNamespace) -> Definition {
+        // `host_key` defaults to the `log_schema().host_key()` if it's not configured in the source.
+        let host_key = self
+            .host_key
+            .clone()
+            .map_or_else(
+                || parse_value_path(log_schema().host_key()).ok(),
+                |k| k.path,
+            )
+            .map(LegacyKey::Overwrite);
+
+        let pid_key = self
+            .pid_key
+            .clone()
+            .and_then(|k| k.path)
+            .map(LegacyKey::Overwrite);
+
+        // There is a global and per-source `log_namespace` config.
+        // The source config overrides the global setting and is merged here.
+        BytesDeserializerConfig
+            .schema_definition(log_namespace)
+            .with_standard_vector_source_metadata()
+            .with_source_metadata(
+                InternalLogsConfig::NAME,
+                host_key,
+                &owned_value_path!("host"),
+                Kind::bytes().or_undefined(),
+                Some("host"),
+            )
+            .with_source_metadata(
+                InternalLogsConfig::NAME,
+                pid_key,
+                &owned_value_path!("pid"),
+                Kind::bytes().or_undefined(),
```

Comment:
Same as above... and would have guessed without looking it would be an integer but 🤷

Reply:
yes to both!